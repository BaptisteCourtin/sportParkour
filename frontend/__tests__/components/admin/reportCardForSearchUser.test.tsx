import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ReportCardForSearchUser from "@/components/admin/reportCardForSearchUser";

describe("ReportCardForSearchUser", () => {
  const mockReportUser = {
    id: "123",
    firstname: "John",
    name: "Doe",
    imageProfill: null,
    nbReportValide: 3,
    nbReportAjoute: 5,
  };

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ReportCardForSearchUser reportUser={mockReportUser} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    render(<ReportCardForSearchUser reportUser={mockReportUser} />);

    // Vérifier la présence du nom complet de l'utilisateur
    expect(screen.getByText("John Doe")).toBeInTheDocument();

    // Vérifier la présence des informations sur les reports
    expect(
      screen.getByText("L'utilisateur a 3 report valide contre lui")
    ).toBeInTheDocument();
    expect(
      screen.getByText("L'utilisateur a émis 5 reports")
    ).toBeInTheDocument();

    // Vérifier que l'image par défaut est affichée
    expect(screen.getByRole("img")).toHaveAttribute("src", "/userDefault.png");
  });

  it("renders user profile image when available", () => {
    const userWithImage = {
      ...mockReportUser,
      imageProfill: "/profile.png",
    };

    render(<ReportCardForSearchUser reportUser={userWithImage} />);

    // Vérifier que l'image de profil est affichée
    expect(screen.getByRole("img")).toHaveAttribute("src", "/profile.png");
  });
});
