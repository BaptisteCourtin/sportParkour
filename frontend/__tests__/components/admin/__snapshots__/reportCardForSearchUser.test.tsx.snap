// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportCardForSearchUser matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="nbRep-3 reportCardForSearchUser"
  >
    <div
      class="userProfil"
    >
      <img
        class="imgProfil"
        src="/userDefault.png"
      />
      <a
        href="/admin/reports/user/123"
      >
        John Doe
      </a>
    </div>
    <p>
      L'utilisateur a 3 report valide contre lui
    </p>
    <p>
      L'utilisateur a Ã©mis 5 reports
    </p>
  </li>
</DocumentFragment>
`;
